Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/baseprograms/BaseAutonomous.java

```diff
@@ -169,15 +400,24 @@ class PoseAndAction {
         }
     }
 
-   /* Booleans 'isRed' (red or blue side), 'isFar' (far or close to backdrop)
-    'location' (center, middle, or right), and 'intake' (Action for use).
-    */
-    PoseAndAction getDriveAction(boolean isRed, boolean isFar, SpikeMarks location, Action intake) {
+    /* Booleans 'isRed' (red or blue side), 'isFar' (far or close to backdrop)
+     'location' (center, middle, or right), and 'intake' (Action for use).
+     */
+    PoseAndAction getDriveAction(boolean isRed, boolean isFar, SpikeMarks location, Action intake, Action moveArm, Action moveDumper, Action startMoveBackward, Action endMoveBackward, Action moveGateAction) {
```

Comment:
Wow, that's a lot of parameters! At this point, I'd recommend creating a new class that holds everything from `isRed` to `moveGateAction` and pass an object of that class as the only parameter for `getDriveAction`.